### PR TITLE
feat: Remove usePrefix and useSuffix properties

### PR DESCRIPTION
**Pull request by talyssoncastro**

### CODE REVIEW

File: cli/src/test/java/com/avioconsulting/mule/cli/DeployerCommandLineTest.groovy [hunk 1]
```diff
@@ -137,7 +135,7 @@ muleDeploy {
         assert actualApp instanceof OnPremDeploymentRequest
         actualApp.with {
             assertThat it.appName.normalizedAppName,
-                       is(equalTo('the-app'))
+                       is(equalTo('avi-the-app-xxx'))
```
Comment (manikmagar) [inline cli/src/test/java/com/avioconsulting/mule/cli/DeployerCommandLineTest.groovy:138]: @talyssoncastro tests that were having `usePrefix false`, should be updated to set `prefix ''`  to retain the original behavior of the test case. Similiarly for `useSuffix`. Also, ensuring that existing tests are covering positive and negative scenarios. If anything is missing, then adding a new oen.

File: library/src/main/java/com/avioconsulting/mule/deployment/api/models/deployment/ApplicationName.groovy [hunk 2]
```diff
@@ -11,10 +11,10 @@ class ApplicationName {
     private static final String EMPTY_STRING=''
     private static final MAX_SIZE_APPLICATION_NAME = 42
 
-    ApplicationName(String baseAppName, Boolean usePrefix, Boolean useSuffix, String prefix, String suffix) {
+    ApplicationName(String baseAppName, String prefix, String suffix) {
         this.baseAppName = baseAppName
-        this.usePrefix = usePrefix
-        this.useSuffix = useSuffix
+        this.usePrefix = prefix != null && !prefix.isBlank()
+        this.useSuffix = suffix != null && !suffix.isBlank()
```
Comment (manikmagar) [inline library/src/main/java/com/avioconsulting/mule/deployment/api/models/deployment/ApplicationName.groovy:17]: Do we still need to set it here? If properties are still on ApplicationName, groovy will still let you set them in the config.

Comment (manikmagar) [reply library/src/main/java/com/avioconsulting/mule/deployment/api/models/deployment/ApplicationName.groovy:17]: The parameters are removed from constructor arguments and properties are marked as private, so they won't be accessible from config.

**Review by manikmagar — APPROVED** [review]
